src/fondant/core/component_spec.py

```diff
@@ -191,38 +163,28 @@ def tags(self) -> t.List[str]:
 
     @property
     def index(self):
-        return ComponentSubset({"fields": {}})
+        return Field(name="index", location=self._specification["index"].location)
```

Comment:
I don't think the index has a location in the component spec?

Reply:
I think we don't need the index in the ComponentSpec anymore.

Reply:
I think you're right.